Framework/PythonInterface/plugins/algorithms/CalculateEfficiencyCorrection.py

```diff
@@ -0,0 +1,256 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import absolute_import, division, print_function
+import numpy as np
+
+from mantid.simpleapi import \
+    CloneWorkspace, ConvertFromDistribution, ConvertToPointData, \
+    ConvertUnits, CreateWorkspace, Rebin, SetSampleMaterial
+from mantid.api import \
+    AlgorithmFactory, CommonBinsValidator, PropertyMode, PythonAlgorithm, \
+    MatrixWorkspaceProperty
+from mantid.kernel import \
+    Direction, FloatBoundedValidator, MaterialBuilder, \
+    StringListValidator, StringMandatoryValidator
+
+TABULATED_WAVELENGTH = 1.7982
+
+
+class CalculateEfficiencyCorrection(PythonAlgorithm):
+    _input_ws = None
+    _output_ws = None
+
+    def category(self):
+        return 'CorrectionFunctions\\EfficiencyCorrections'
+
+    def name(self):
+        return 'CalculateEfficiencyCorrection'
+
+    def summary(self):
+        return 'Calculate an efficiency correction using various inputs. Can be used to determine \
+                an incident spectrum after correcting a measured spectrum from beam monitors \
+                or vanadium measurements.'
+
+    def seeAlso(self):
+        return [ "He3TubeEfficiency", "CalculateSampleTransmission",
+                 "DetectorEfficiencyCor", "DetectorEfficiencyCorUser",
+                 "CalculateEfficiency", "ComputeCalibrationCoefVan" ]
+
+    def PyInit(self):
+        self.declareProperty(
+            MatrixWorkspaceProperty('InputWorkspace', '',
+                                    direction=Direction.Input,
+                                    optional=PropertyMode.Optional,
+                                    validator=CommonBinsValidator()),
+            doc='Input workspace with wavelength range to calculate for the correction.')
+
+        self.declareProperty(name='WavelengthRange', defaultValue='',
+                             doc='Wavelength range to calculate efficiency for.')
+
+        self.declareProperty(
+            MatrixWorkspaceProperty('OutputWorkspace', '',
+                                    direction=Direction.Output),
+            doc="Outputs the efficiency correction which can be applied by multiplying \
+                 the OutputWorkspace to the workspace that requires the correction.")
+
+        self.declareProperty(
+            name='ChemicalFormula', defaultValue='None',
+            validator=StringMandatoryValidator(),
+            doc='Sample chemical formula used to determine :math:`\\sigma (\\lambda_{ref})` term.')
```

Comment:
The math formatting looks very nice in the html documentation but does not work well with the tooltips in the automatically generated dialog where no formatting is done. I would recommend leaving the math stuff out of the doc strings here and explain the properties more rigorously in the `.rst` file. There are similar issues in the other properties below too.